scripts/python/Ne_from_SLiM_vcfs.py

```diff
@@ -71,32 +66,37 @@ def create_sfs_dict(inpath):
 def write_thetaNe_values(sfs_dict, outpath):
 
     # Open csv to write resutls
-    outfile = open(outpath, 'w+')
-
-    # Write header
-    outfile.write('N,bot,gen,theta_pi,theta_w,Ne_pi,Ne_w\n')
-
-    mu = float(1e-8)
-
-    # Interate through sfs dictionary and write to csv
-    for key in sfs_dict.keys():
-
-        # Get dict values
-        d = sfs_dict[key]
-
-        l = d.keys()
-        l = sorted(l)
-
-        # Iterate through nested dictionary values (generations)
-        for g in l:
-            pi = d[g].theta_pi()
-            watersons = d[g].theta_w()
-            Ne_pi = (pi / (4 * mu))
-            Ne_w = (watersons / (4 * mu))
-#             print(g, d[g].theta_pi(), d[g].theta_w(),Ne_pi,Ne_w)
-            N_and_Gen = key.split('-')
-            outfile.write('{0},{1},{2},{3},{4},{5},{6}\n'.format(N_and_Gen[0], N_and_Gen[1], g, pi,watersons, Ne_pi, Ne_w))
-    outfile.close()
+    with open(outpath, 'w+') as f:
+
+        # Write header
+        f.write('N,bot,gen,theta_pi,theta_w,Ne_pi,Ne_w\n')
+
+        mu = float(1e-8)
+
+        # Interate through sfs dictionary and write to csv
+        for key in sfs_dict.keys():
+
+            # Get dict values, which are encoded as <pop size>-<bottleneck>
+            size_bot = sfs_dict[key]
+
+            # Get generations as list
+            generations = sorted(size_bot.keys())
+
+            # Iterate through nested dictionary values (generations)
+            for gen in generations:
+                pi = size_bot[gen].theta_pi()
+                watersons = size_bot[gen].theta_w()
+                Ne_pi = round((pi / (4 * mu)), 3)
+                Ne_w = round((watersons / (4 * mu)), 3)
+                pop_size = key.split('-')[0]
+                bottleneck = key.split('-')[1]
+                f.write('{0},{1},{2},{3},{4},{5},{6}\n'.format(pop_size,
+                                                               bottleneck,
+                                                               gen,
+                                                               pi,
+                                                               watersons,
+                                                               Ne_pi,
+                                                               Ne_w))
 
 
 inpath = "../../data/test_SLiM_output/"
```

Comment:
Doesn't need to be done now but could be a future cleanup goal to remove all hardcoded paths and use `argparse` with inpath/outpath args

Reply:
Done. Output directories are created if they don't exist. Eventually, we might consider creating an actual "project" with an `__init__.py` and `setup.py`. Might make path management stuff a bit easier.